universal-application-tool-0.0.1/tailwind.config.js

```diff
@@ -1,42 +1,105 @@
+const fs = require('fs')
+
+// For stylesDict from Styles.java and ReferenceClasses.java
+const RGX_KEY = /(?<= +public +static +final +String +)([0-9A-Z_]+)/g
+const RGX_VAL =
+  /(?<= +public +static +final +String +[0-9A-Z_]+ += +")([a-z0-9-/.]+)/g
+
+const STYLE_RGX = /(?<=(Styles|ReferenceClasses)\.)([0-9A-Z_]+)/g
+
+const PREFIXES = [
+  'even',
+  'focus',
+  'focus-within',
+  'hover',
+  'disabled',
+  'sm',
+  'md',
+  'lg',
+  'xl',
+  '2xl',
+]
+
+// Used to read Styles.java and ReferenceClasses.java to get dictionary mapping
+// for all possible base styles (no prefixes)
+function addStyleDictMatches(matches, file_contents) {
+  let count = 1
+  for (const line of file_contents) {
+    let match_key = line.match(RGX_KEY)
+    let match_val = line.match(RGX_VAL)
+
+    // Both 'variable' and tailwind str are probably on same line
+    // Even though java probably doesn't require it
+    if (Array.isArray(match_key) && Array.isArray(match_val)) {
+      if (match_key.length === 1 && match_val.length === 1) {
+        matches[match_key[0]] = match_val[0]
+      } else {
+        throw "strange line in 'Styles.java' at line " + count.toString()
+      }
+    }
+
+    count++
+  }
+}
+
+function getStylesDict() {
+  const matches = {}
+  let folder = './app/views/style/'
+  try {
+    let specialFiles = ['Styles.java', 'ReferenceClasses.java']
```

Comment:
Suggestion: make this a file level variable next to the RGX that references them too, then their relationship is clearer and updating them will be less error prone.